[Bug]: openapi-generator 7.4.0 fails to resolve refs to files

### Package name

schematics

### Package version

11.0.1

### Reproduction steps

_You can ask me in Teams and I'll provide a reproduction .zip_

__Step 1: create models__
```shell
OTTER_ROOT=todefine
mkdir ama-sdk-repro && cd ama-sdk-repro && mkdir models
cp $OTTER_ROOT/packages/@ama-sdk/showcase-sdk/openapi.yml ./models/pet-specs-untouched.yaml
cp ./models/pet-specs-untouched.yaml ./models/pet-specs-split.yaml
touch ./models/pet-response.yaml
```

__Step 2: Manually modify ``pet-specs-split.yaml``__

Modify the ``paths./pet.post.responses.'200'`` to point to a ``components.response`` ``ok-pet``:
```yaml
  /pet:
    post:
      tags:
        - pet
      summary: Add a new pet to the store
      description: Add a new pet to the store
      operationId: addPet
      responses:
        '200':
          $ref: '#/components/responses/ok-pet'
```
Create the ``components.response`` ``ok-pet``:
```yaml
  responses:
    ok-pet:
      description: Successful Pet operation
      content:
        application/xml:
          schema:
            $ref: './pet-response.yaml'
        application/json:
          schema:
            $ref: './pet-response.yaml'
```

In ``pet-response.yaml``, modify the refs to point to ``pet-specs-split.yaml#/components``. Also add a title:
```yaml
x-swagger-router-model: io.swagger.petstore.model.Pet
title: Pet
type: object
required:
  - name
  - photoUrls
properties:
  id:
    type: integer
    format: int64
    example: 10
  name:
    type: string
    example: doggie
  category:
    $ref: './pet-specs-split.yaml#/components/schemas/Category'
  photoUrls:
    type: array
    xml:
      wrapped: true
    items:
      type: string
      xml:
        name: photoUrl
  tags:
    type: array
    xml:
      wrapped: true
    items:
      $ref: './pet-specs-split.yaml#/components/schemas/Tag'
      xml:
        name: tag
  status:
    type: string
    description: pet status in the store
    enum:
      - available
      - pending
      - sold
xml:
  name: pet
```

__Step 3: generate SDKs for version 9.6.x and 11.0.x__
```bash
npm create @ama-sdk@9.6.3 typescript sdk-otter-9 -- --spec-path ./models/pet-specs-untouched.yaml
npm create @ama-sdk typescript sdk-otter-11 -- --spec-path ./models/pet-specs-untouched.yaml
```
_Note_: I had troubles with ``--spec-path``  and ``--generator-key`` on v11 and had to manually modify the scripts in ``package.json``.

Also, I've duplicated the ``spec:regen`` script so that it targets both split & untouched specs:
```text
// v9
"spec:regen:untouched": "npm run generate -- --spec-path ../models/pet-specs-untouched.yaml && npm run clear-index",
"spec:regen:split": "npm run generate -- --spec-path ../models/pet-specs-split.yaml && npm run clear-index",

// v11
"spec:regen:untouched": "npm run generate -- --spec-path ../models/pet-specs-untouched.yaml --generator-key sdk-otter-11 && amasdk-clear-index",
"spec:regen:split": "npm run generate -- --spec-path ../models/pet-specs-split.yaml --generator-key sdk-otter-11 && amasdk-clear-index",
```

### Current result

- ``npm run spec:regen:untouched`` works for both v9 and v11
- ``npm run spec:regen:split`` works for v9
- ``npm run spec:regen:split`` fails for v11 with below exception

```log
Did set selected version to 7.4.0
[main] WARN  io.swagger.v3.parser.OpenAPIV3Parser - Exception while resolving:
java.lang.RuntimeException: Unable to load RELATIVE ref: ./pet-response.yaml path: /home/fhamelin/git_clones/ama-sdk-repro/sdk-otter-11
        at io.swagger.v3.parser.util.RefUtils.readExternalRef(RefUtils.java:220)
        at io.swagger.v3.parser.ResolverCache.loadRef(ResolverCache.java:150)
        at io.swagger.v3.parser.processors.ExternalRefProcessor.processRefToExternalSchema(ExternalRefProcessor.java:88)
        at io.swagger.v3.parser.processors.SchemaProcessor.processReferenceSchema(SchemaProcessor.java:236)
        at io.swagger.v3.parser.processors.SchemaProcessor.processSchema(SchemaProcessor.java:60)
        at io.swagger.v3.parser.processors.ResponseProcessor.processResponse(ResponseProcessor.java:56)
        at io.swagger.v3.parser.processors.OperationProcessor.processOperation(OperationProcessor.java:86)
        at io.swagger.v3.parser.processors.PathsProcessor.processPaths(PathsProcessor.java:88)
        at io.swagger.v3.parser.OpenAPIResolver.resolve(OpenAPIResolver.java:72)
        at io.swagger.v3.parser.OpenAPIResolver.resolve(OpenAPIResolver.java:59)
        at io.swagger.v3.parser.OpenAPIV3Parser.resolve(OpenAPIV3Parser.java:238)
        at io.swagger.v3.parser.OpenAPIV3Parser.readContents(OpenAPIV3Parser.java:181)
        at io.swagger.v3.parser.OpenAPIV3Parser.readLocation(OpenAPIV3Parser.java:97)
        at io.swagger.parser.OpenAPIParser.readLocation(OpenAPIParser.java:16)
        at org.openapitools.codegen.config.CodegenConfigurator.toContext(CodegenConfigurator.java:686)
        at org.openapitools.codegen.config.CodegenConfigurator.toClientOptInput(CodegenConfigurator.java:744)
        at org.openapitools.codegen.cmd.Generate.execute(Generate.java:527)
        at org.openapitools.codegen.cmd.OpenApiGeneratorCommand.run(OpenApiGeneratorCommand.java:32)
        at org.openapitools.codegen.OpenAPIGenerator.main(OpenAPIGenerator.java:66)
Caused by: java.lang.RuntimeException: Could not find ./pet-response.yaml on the classpath
        at io.swagger.v3.parser.util.ClasspathHelper.loadFileFromClasspath(ClasspathHelper.java:33)
        at io.swagger.v3.parser.util.RefUtils.readExternalRef(RefUtils.java:214)
        ... 18 common frames omitted
[main] WARN  o.o.codegen.utils.ModelUtils - Failed to get the schema name: ./pet-response.yaml
[main] WARN  o.o.codegen.utils.ModelUtils - Failed to get the schema name: ./pet-response.yaml
[main] WARN  o.o.codegen.utils.ModelUtils - Failed to get the schema name: ./pet-response.yaml
[main] WARN  o.o.codegen.utils.ModelUtils - Failed to get the schema name: ./pet-response.yaml
Exception in thread "main" org.openapitools.codegen.SpecValidationException: There were issues with the specification. The option can be disabled via validateSpec (Maven/Gradle) or --skip-validate-spec (CLI).
 | Error count: 1, Warning count: 2
Errors:
        -Unable to load RELATIVE ref: ./pet-response.yaml path: /home/fhamelin/git_clones/ama-sdk-repro/sdk-otter-11
Warnings:
        -Unable to load RELATIVE ref: ./pet-response.yaml path: /home/fhamelin/git_clones/ama-sdk-repro/sdk-otter-11

        at org.openapitools.codegen.config.CodegenConfigurator.toContext(CodegenConfigurator.java:717)
        at org.openapitools.codegen.config.CodegenConfigurator.toClientOptInput(CodegenConfigurator.java:744)
        at org.openapitools.codegen.cmd.Generate.execute(Generate.java:527)
        at org.openapitools.codegen.cmd.OpenApiGeneratorCommand.run(OpenApiGeneratorCommand.java:32)
        at org.openapitools.codegen.OpenAPIGenerator.main(OpenAPIGenerator.java:66)
Error: OpenApiGeneratorCli failed to run OpenApiGenerator with command 'openapi-generator-cli generate --custom-generator=/home/fhamelin/git_clones/ama-sdk-repro/sdk-otter-11/node_modules/@ama-sdk/schematics/schematics/typescript/core/openapi-codegen-typescript/target/typescriptFetch-openapi-generator.jar -g typescriptFetch -i open-api.yaml -o .
```

### Expected result

``npm run spec:regen:split`` should work on v11 (and v10 which I haven't tried)

### Additional comments

- Using JDK ``11.0.14.1-jbr``
- For the failing case, it uses openapi-generator ``7.4.0``. Tried to upgrade to openapi-generator ``7.7.0`` but same issue
- The working case uses openapi-generator ``6.6.0``
- When generating the ``typescriptFetch-openapi-generator.jar`` for ``7.7.0``, I had to modify the file ``AbstractTypeScriptClientCodegen`` because they moved the ``getSchemaItems`` method to ``ModelUtils`` (see [DefaultCodegen.java](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java)). Note that the exception above has warnings for ``ModelUtils``, so it could be related.
- This kind of issue seems to be reccurrent with openapi-generator. It gets fixed then breaks again:
https://github.com/OpenAPITools/openapi-generator/issues/17554
https://github.com/OpenAPITools/openapi-generator/issues/18429

--- a/packages/@ama-sdk/schematics/schematics/typescript/core/helpers/copy-referenced-files.spec.ts
+++ b/packages/@ama-sdk/schematics/schematics/typescript/core/helpers/copy-referenced-files.spec.ts
@@ -2,11 +2,11 @@ import { cleanVirtualFileSystem, useVirtualFileSystem } from '@o3r/test-helpers'
 import { readFile } from 'node:fs/promises';
 import { dirname, join } from 'node:path';
 
-describe('Copy Referenced Files', () => {
+describe('Specs processing', () => {
   const virtualFileSystem = useVirtualFileSystem();
-  const copyReferencedFiles = require('./copy-referenced-files').copyReferencedFiles;
+  const {copyReferencedFiles, updateLocalRelativeRefs} = require('./copy-referenced-files');
 
-  const migrationScriptMocksPath = join(__dirname, '../../../../testing/mocks');
+  const specsMocksPath = join(__dirname, '../../../../testing/mocks');
   const specFilePath = '../models/split-spec/split-spec.yaml';
   const outputDirectory = './local-references';
 
@@ -14,7 +14,7 @@ describe('Copy Referenced Files', () => {
     if (!virtualFileSystem.existsSync(dirname(virtualPath))) {
       await virtualFileSystem.promises.mkdir(dirname(virtualPath), {recursive: true});
     }
-    await virtualFileSystem.promises.writeFile(virtualPath, await readFile(join(migrationScriptMocksPath, realPath), {encoding: 'utf8'}));
+    await virtualFileSystem.promises.writeFile(virtualPath, await readFile(join(specsMocksPath, realPath), {encoding: 'utf8'}));
   };
 
   beforeAll(async () => {
@@ -38,5 +38,16 @@ describe('Copy Referenced Files', () => {
     expect(virtualFileSystem.existsSync(join(outputDirectory, 'spec-chunk2.yaml'))).toBe(true);
     expect(virtualFileSystem.existsSync(join(outputDirectory, 'spec-chunk3/spec-chunk3.yaml'))).toBe(true);
     expect(virtualFileSystem.existsSync(join(outputDirectory, 'spec-chunk4/spec-chunk4.yaml'))).toBe(true);
+  });
+
+  it('should update with new local basepath', async () => {
+    const specWitheRelativesFilePath = 'split-spec/split-spec.yaml';
+    const expectedSpecWitheRelativesFilePath = 'split-spec/spec-with-updated-paths.yaml';
+    const expectedContent = await readFile(join(specsMocksPath, expectedSpecWitheRelativesFilePath), {encoding: 'utf8'});
+    const specContent = await readFile(join(specsMocksPath, specWitheRelativesFilePath), {encoding: 'utf8'});
+
+    const baseRelativePath = await copyReferencedFiles(specFilePath, './output-local-directory');
+    const newSpecContent = await updateLocalRelativeRefs(specContent, baseRelativePath);
+    expect(newSpecContent).toBe(expectedContent);
   });
 });

--- a/packages/@ama-sdk/schematics/schematics/typescript/core/helpers/copy-referenced-files.spec.ts
+++ b/packages/@ama-sdk/schematics/schematics/typescript/core/helpers/copy-referenced-files.spec.ts
@@ -1,0 +1,42 @@
+import { cleanVirtualFileSystem, useVirtualFileSystem } from '@o3r/test-helpers';
+import { readFile } from 'node:fs/promises';
+import { dirname, join } from 'node:path';
+
+describe('Copy Referenced Files', () => {
+  const virtualFileSystem = useVirtualFileSystem();
+  const copyReferencedFiles = require('./copy-referenced-files').copyReferencedFiles;
+
+  const migrationScriptMocksPath = join(__dirname, '../../../../testing/mocks');
+  const specFilePath = '../models/split-spec/split-spec.yaml';
+  const outputDirectory = './local-references';
+
+  const copyMockFile = async (virtualPath: string, realPath: string) => {
+    if (!virtualFileSystem.existsSync(dirname(virtualPath))) {
+      await virtualFileSystem.promises.mkdir(dirname(virtualPath), {recursive: true});
+    }
+    await virtualFileSystem.promises.writeFile(virtualPath, await readFile(join(migrationScriptMocksPath, realPath), {encoding: 'utf8'}));
+  };
+
+  beforeAll(async () => {
+    await virtualFileSystem.promises.mkdir(dirname(specFilePath), {recursive: true});
+    await copyMockFile(specFilePath, 'split-spec/split-spec.yaml');
+    await copyMockFile('../models/split-spec/spec-chunk1.yaml', 'split-spec/spec-chunk1.yaml');
+    await copyMockFile('../models/spec-chunk2.yaml', 'spec-chunk2.yaml');
+    await copyMockFile('../models/spec-chunk3/spec-chunk3.yaml', 'spec-chunk3/spec-chunk3.yaml');
+    await copyMockFile('../models/spec-chunk4/spec-chunk4.yaml', 'spec-chunk4/spec-chunk4.yaml');
+  });
+
+  afterAll(() => {
+    cleanVirtualFileSystem();
+  });
+
+  it('should copy the local files referenced in the spec', async () => {
+    const baseRelativePath = await copyReferencedFiles(specFilePath, outputDirectory);
+    expect(baseRelativePath).toMatch(/^local-references[\\/]split-spec$/);
+    expect(virtualFileSystem.existsSync(join(outputDirectory, 'split-spec/split-spec.yaml'))).toBe(true);
+    expect(virtualFileSystem.existsSync(join(outputDirectory, 'split-spec/spec-chunk1.yaml'))).toBe(true);
+    expect(virtualFileSystem.existsSync(join(outputDirectory, 'spec-chunk2.yaml'))).toBe(true);
+    expect(virtualFileSystem.existsSync(join(outputDirectory, 'spec-chunk3/spec-chunk3.yaml'))).toBe(true);
+    expect(virtualFileSystem.existsSync(join(outputDirectory, 'spec-chunk4/spec-chunk4.yaml'))).toBe(true);
+  });
+});

--- a/packages/@ama-sdk/schematics/schematics/typescript/core/helpers/copy-referenced-files.ts
+++ b/packages/@ama-sdk/schematics/schematics/typescript/core/helpers/copy-referenced-files.ts
@@ -1,6 +1,6 @@
 import { existsSync } from 'node:fs';
 import { copyFile, mkdir, readFile, rm } from 'node:fs/promises';
-import { dirname, join, normalize, posix, relative, resolve } from 'node:path';
+import { dirname, join, normalize, posix, relative, resolve, sep } from 'node:path';
 
 const refMatcher = /\B['"]?[$]ref['"]?\s*:\s*([^#\n]+)/g;
 
@@ -53,7 +53,7 @@ export function updateLocalRelativeRefs(specContent: string, newBaseRelativePath
   return specContent.replace(refMatcher, (match, ref: string) => {
     const refPath = ref.replace(/['"]/g, '');
     return refPath.startsWith('.') ?
-      match.replace(refPath, formatPath(normalize(posix.join(newBaseRelativePath, refPath))))
+      match.replace(refPath, formatPath(normalize(posix.join(newBaseRelativePath.replaceAll(sep, posix.sep), refPath))))
       : match;
   });
 }
@@ -64,12 +64,9 @@ export function updateLocalRelativeRefs(specContent: string, newBaseRelativePath
  * @param outputDirectory
  */
 export async function copyReferencedFiles(specFilePath: string, outputDirectory: string) {
-  const dedupe = (paths: string[]) =>
-    paths.filter((refPath, index) => {
-      const actualPath = join(dirname(specFilePath), refPath);
-      return paths.findIndex((otherRefPath) => join(dirname(specFilePath), otherRefPath) === actualPath) === index;
-    });
-  const refPaths = dedupe(await extractRefPathRecursive(specFilePath, specFilePath, new Set()));
+  const dedupe = (paths: string[]) => ([...new Set(paths)]);
+  const allRefPaths = await extractRefPathRecursive(specFilePath, specFilePath, new Set());
+  const refPaths = dedupe(allRefPaths);
   if (refPaths.length) {
     if (existsSync(outputDirectory)) {
       await rm(outputDirectory, { recursive: true });

--- a/packages/@ama-sdk/schematics/schematics/typescript/core/helpers/copy-referenced-files.ts
+++ b/packages/@ama-sdk/schematics/schematics/typescript/core/helpers/copy-referenced-files.ts
@@ -1,0 +1,96 @@
+import { existsSync } from 'node:fs';
+import { copyFile, mkdir, readFile, rm } from 'node:fs/promises';
+import { dirname, join, normalize, posix, relative, resolve } from 'node:path';
+
+const refMatcher = /\B['"]?[$]ref['"]?\s*:\s*([^#\n]+)/g;
+
+/**
+ * Extract the list of local references from a single spec file content
+ * @param specContent
+ * @param basePath
+ */
+function extractRefPaths(specContent: string, basePath: string): string[] {
+  const refs = specContent.match(refMatcher);
+  return refs ?
+    refs
+      .map((capture) => capture.replace(refMatcher, '$1').replace(/['"]/g, ''))
+      .filter((refPath) => refPath.startsWith('.'))
+      .map((refPath) => join(basePath, refPath))
+    : [];
+}
+
+/**
+ * Recursively extract the list of local references starting from the input spec file
+ * @param specFilePath
+ * @param referenceFilePath
+ * @param visited
+ */
+async function extractRefPathRecursive(specFilePath: string, referenceFilePath: string, visited: Set<string>): Promise<string[]> {
+  const resolvedFilePath = resolve(specFilePath);
+  if (!visited.has(resolvedFilePath)) {
+    visited.add(resolvedFilePath);
+
+    const specContent = await readFile(specFilePath, {encoding: 'utf8'});
+    const refPaths = extractRefPaths(specContent, relative(dirname(referenceFilePath), dirname(specFilePath)));
+    const recursiveRefPaths = await Promise.all(
+      refPaths.map((refPath) => extractRefPathRecursive(join(dirname(referenceFilePath), refPath), referenceFilePath, visited))
+    );
+    return [
+      ...refPaths,
+      ...recursiveRefPaths.flat()
+    ];
+  }
+  return [];
+}
+
+/**
+ * Replace all the local relative references using the new base relative path
+ * @param specContent
+ * @param newBaseRelativePath
+ */
+export function updateLocalRelativeRefs(specContent: string, newBaseRelativePath: string) {
+  const formatPath = (inputPath:string) => (inputPath.startsWith('.') ? inputPath : `./${inputPath}`).replace(/\\+/g, '/');
+  return specContent.replace(refMatcher, (match, ref: string) => {
+    const refPath = ref.replace(/['"]/g, '');
+    return refPath.startsWith('.') ?
+      match.replace(refPath, formatPath(normalize(posix.join(newBaseRelativePath, refPath))))
+      : match;
+  });
+}
+
+/**
+ * Copy the local files referenced in the input spec file to the output directory
+ * @param specFilePath
+ * @param outputDirectory
+ */
+export async function copyReferencedFiles(specFilePath: string, outputDirectory: string) {
+  const dedupe = (paths: string[]) =>
+    paths.filter((refPath, index) => {
+      const actualPath = join(dirname(specFilePath), refPath);
+      return paths.findIndex((otherRefPath) => join(dirname(specFilePath), otherRefPath) === actualPath) === index;
+    });
+  const refPaths = dedupe(await extractRefPathRecursive(specFilePath, specFilePath, new Set()));
+  if (refPaths.length) {
+    if (existsSync(outputDirectory)) {
+      await rm(outputDirectory, { recursive: true });
+    }
+
+    // Calculate the lowest level base path to keep the same directory structure
+    const maxDepth = Math.max(...refPaths.map((refPath) => refPath.split('..').length));
+    const basePath = join(specFilePath, '../'.repeat(maxDepth));
+    const baseRelativePath = relative(basePath, dirname(specFilePath));
+
+    // Copy the files
+    await Promise.all(refPaths.map(async (refPath) => {
+      const sourcePath = join(dirname(specFilePath), refPath);
+      const destPath = join(outputDirectory, baseRelativePath, refPath);
+      if (!existsSync(dirname(destPath))) {
+        await mkdir(dirname(destPath), { recursive: true });
+      }
+      await copyFile(sourcePath, destPath);
+    }));
+
+    return join(outputDirectory, baseRelativePath);
+  }
+  return '';
+}

--- a/packages/@ama-sdk/schematics/schematics/typescript/core/index.ts
+++ b/packages/@ama-sdk/schematics/schematics/typescript/core/index.ts
@@ -24,6 +24,7 @@ import { OpenApiCliOptions } from '../../code-generator/open-api-cli-generator/o
 import { treeGlob } from '../../helpers/tree-glob';
 import { NgGenerateTypescriptSDKCoreSchematicsSchema } from './schema';
 import { OpenApiCliGenerator } from '../../code-generator/open-api-cli-generator/open-api-cli.generator';
+import { copyReferencedFiles, updateLocalRelativeRefs } from './helpers/copy-referenced-files';
 import { generateOperationFinderFromSingleFile } from './helpers/path-extractor';
 
 const JAVA_OPTIONS = ['specPath', 'specConfigPath', 'globalProperty', 'outputPath'];
@@ -153,10 +154,19 @@ function ngGenerateTypescriptSDKFn(options: NgGenerateTypescriptSDKCoreSchematic
     let specContent!: string;
     if (URL.canParse(generatorOptions.specPath) && (new URL(generatorOptions.specPath)).protocol.startsWith('http')) {
       specContent = await (await fetch(generatorOptions.specPath)).text();
+      specContent = updateLocalRelativeRefs(specContent, path.dirname(generatorOptions.specPath));
     } else {
       const specPath = path.isAbsolute(generatorOptions.specPath) || !options.directory ?
         generatorOptions.specPath : path.join(options.directory, generatorOptions.specPath);
       specContent = readFileSync(specPath, {encoding: 'utf-8'}).toString();
+
+      if (path.relative(process.cwd(), specPath).startsWith('..')) {
+        // TODO would be better to create files on tree instead of FS
+        const newRelativePath = await copyReferencedFiles(specPath, './spec-local-references');
+        if (newRelativePath) {
+          specContent = updateLocalRelativeRefs(specContent, newRelativePath);
+        }
+      }
     }
 
     try {

--- a/packages/@ama-sdk/schematics/schematics/typescript/core/index.ts
+++ b/packages/@ama-sdk/schematics/schematics/typescript/core/index.ts
@@ -162,6 +162,7 @@ function ngGenerateTypescriptSDKFn(options: NgGenerateTypescriptSDKCoreSchematic
 
       if (path.relative(process.cwd(), specPath).startsWith('..')) {
         // TODO would be better to create files on tree instead of FS
+        // https://github.com/AmadeusITGroup/otter/issues/2078
         const newRelativePath = await copyReferencedFiles(specPath, './spec-local-references');
         if (newRelativePath) {
           specContent = updateLocalRelativeRefs(specContent, newRelativePath);

--- a/packages/@ama-sdk/schematics/testing/mocks/spec-chunk2.yaml
+++ b/packages/@ama-sdk/schematics/testing/mocks/spec-chunk2.yaml
@@ -1,0 +1,9 @@
+title: Pet
+type: object
+properties:
+  id:
+    type: integer
+    format: int64
+    example: 10
+  category:
+    $ref: './split-spec/split-spec.yaml#/components/schemas/Category'

--- a/packages/@ama-sdk/schematics/testing/mocks/spec-chunk3/spec-chunk3.yaml
+++ b/packages/@ama-sdk/schematics/testing/mocks/spec-chunk3/spec-chunk3.yaml
@@ -1,0 +1,9 @@
+title: Pet
+type: object
+properties:
+  id:
+    type: integer
+    format: int64
+    example: 10
+  category:
+    $ref: '../spec-chunk4/spec-chunk4.yaml#/components/schemas/Category'

--- a/packages/@ama-sdk/schematics/testing/mocks/spec-chunk4/spec-chunk4.yaml
+++ b/packages/@ama-sdk/schematics/testing/mocks/spec-chunk4/spec-chunk4.yaml
@@ -1,0 +1,12 @@
+components:
+  schemas:
+    Category:
+      type: object
+      properties:
+        id:
+          type: integer
+          format: int64
+          example: 10
+        name:
+          type: string
+          example: "test"

--- a/packages/@ama-sdk/schematics/testing/mocks/split-spec/spec-chunk1.yaml
+++ b/packages/@ama-sdk/schematics/testing/mocks/split-spec/spec-chunk1.yaml
@@ -7,3 +7,5 @@ properties:
     example: 10
   category:
     $ref: './split-spec.yaml#/components/schemas/Category'
+  category2:
+      $ref: '../spec-chunk4/spec-chunk4.yaml#/components/schemas/Category'

--- a/packages/@ama-sdk/schematics/testing/mocks/split-spec/spec-chunk1.yaml
+++ b/packages/@ama-sdk/schematics/testing/mocks/split-spec/spec-chunk1.yaml
@@ -1,0 +1,9 @@
+title: Pet
+type: object
+properties:
+  id:
+    type: integer
+    format: int64
+    example: 10
+  category:
+    $ref: './split-spec.yaml#/components/schemas/Category'

--- a/packages/@ama-sdk/schematics/testing/mocks/split-spec/spec-with-updated-paths.yaml
+++ b/packages/@ama-sdk/schematics/testing/mocks/split-spec/spec-with-updated-paths.yaml
@@ -1,0 +1,45 @@
+openapi: 3.0.2
+info:
+  description: test
+  title: test
+  version: 0.0.0
+paths:
+  /test:
+    get:
+      responses:
+        '200':
+          description: test
+          content:
+            application/json:
+              schema:
+                $ref: './output-local-directory/split-spec/spec-chunk1.yaml'
+  /test2:
+    get:
+      responses:
+        '200':
+          description: test
+          content:
+            application/json:
+              schema:
+                $ref: './output-local-directory/spec-chunk2.yaml'
+  /test3:
+    get:
+      responses:
+        '200':
+          description: test
+          content:
+            application/json:
+              schema:
+                $ref: './output-local-directory/spec-chunk3/spec-chunk3.yaml'
+components:
+  schemas:
+    Category:
+      type: object
+      properties:
+        id:
+          type: integer
+          format: int64
+          example: 10
+        name:
+          type: string
+          example: "test"

--- a/packages/@ama-sdk/schematics/testing/mocks/split-spec/split-spec.yaml
+++ b/packages/@ama-sdk/schematics/testing/mocks/split-spec/split-spec.yaml
@@ -1,0 +1,45 @@
+openapi: 3.0.2
+info:
+  description: test
+  title: test
+  version: 0.0.0
+paths:
+  /test:
+    get:
+      responses:
+        '200':
+          description: test
+          content:
+            application/json:
+              schema:
+                $ref: './spec-chunk1.yaml'
+  /test2:
+    get:
+      responses:
+        '200':
+          description: test
+          content:
+            application/json:
+              schema:
+                $ref: '../spec-chunk2.yaml'
+  /test3:
+    get:
+      responses:
+        '200':
+          description: test
+          content:
+            application/json:
+              schema:
+                $ref: '../spec-chunk3/spec-chunk3.yaml'
+components:
+  schemas:
+    Category:
+      type: object
+      properties:
+        id:
+          type: integer
+          format: int64
+          example: 10
+        name:
+          type: string
+          example: "test"

